apps/photos/src/utils/collection/index.ts

```diff
@@ -260,12 +272,27 @@ export function isValidMoveTarget(
 ) {
     return (
         sourceCollectionID !== targetCollection.id &&
-        !isCollectionHidden(targetCollection) &&
+        !isHiddenCollection(targetCollection) &&
         !isQuickLinkCollection(targetCollection) &&
         !isIncomingShare(targetCollection, user)
     );
 }
 
+export function isValidReplacementAlbum(
+    collection: Collection,
+    user: User,
+    wantedCollectionName: string
+) {
+    return (
+        collection.name === wantedCollectionName &&
+        (collection.type === CollectionType.album ||
+            collection.type === CollectionType.folder) &&
+        !isHiddenCollection(collection) &&
```

Comment:
Why add this check?
Suppose if I have a hidden folder on my machine. I drag it and mark it the folder has hidden. Next time when I am dragging the same folder, we shouldn't be creating new album. Adding a check against `defaultHiddenFolder` and `isIncomingShare` should be good enough.

Reply:
yes yes, it does that only, we currently only have default hidden collection and `isHiddenCollection` checks for it is the defaultHiddenCollection. Will update the function name